src/CoreShop/Bundle/CoreBundle/Security/ObjectUserProvider.php

```diff
@@ -67,10 +69,32 @@ public function loadUserByUsername($username)
     {
         $this->checkClass();
 
-        $getter = sprintf('getBy%s', ucfirst($this->usernameField));
+        /** @var Concrete $class */
```

Comment:
Shouldn’t we better inject the repo?

Reply:
I'm not sure about this since this class is a very generic one, see comment on the`className` property:

```
The pimcore class name to be used. 
Needs to be a fully qualified class name
(e.g. Pimcore\Model\DataObject\User or your custom user class
extending the generated one.)
```

Reply:
Just checked again, and we should change that to the Repository. The className variable is always just the Customer Class -> so the repo uses the same class. And therefore we can just use the Repo function which does exactly the same.